### ISSUE TITLE
Dispatcher::resolveController should continue search on non-perfect-match

### ISSUE THREAD

**Issue by eschiendorfer** [thirtybees/thirtybees#1906]
As shortly discussed here https://github.com/thirtybees/thirtybees/commit/7018d9cf29411fef719b30a2ea105b8ba9547599#commitcomment-147841274, the resolveController method does always return a product result, if the "link_rewrite" value is matching. This behaviour is very useful in case a merchant changes his url structure or renames a category. His product links will still work.

BUT this has also the drawback, if another entity (lets say a blog post from a module) uses the same link_rewrite value:

- domain.com/category/homer-simpson
- domain.com/blog/homer-simpson

Even if you call the later domain.com/blog/homer-simpson, the product will be returned. Therefore we should extend the resolveController method for a non-perfect-match.

In the situation above, the method productID() should return a non-perfect-match. If this happens, the resolveController() method continues to search. If he founds anything that matches the url, he returns this entity otherwise the product will be returned (as now).

**Comment by Zengraph** [comment]
You are right, it's very strange behaviour now that when you call a /blog/ url it does return a product url, it should not do that at all, maybe needs to check modules routes before searching for product rewrite.
